Editor/Utils/ScanSolidTexture.cs

```diff
@@ -0,0 +1,214 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using UnityEditor;
+using UnityEngine;
+using Debug = UnityEngine.Debug;
+
+namespace Unity.ProjectAuditor.Editor.Modules
+{
+    /// <summary>
+    /// Scans textures ans check if they are single solid color ones.
+    /// </summary>
+    /// <author>
+    /// Unity Labs SuperScience
+    /// <author>
+    public static class ScanSolidTexture
+    {
+        /// <summary>
+        /// Check if a texture is a single solid color and its size is over than 1x1
+        /// </summary>
+        /// <param name="textureImporter">The texture importer of the texture.</param>
+        /// <param name="texture">The texture to check.</param>
+        /// <returns>True if the texture is a single solid color above 1x1.</returns>
+        public static bool IsTextureSolidColorTooBig(TextureImporter textureImporter, Texture texture)
+        {
+            bool isTooBig = false;
+            bool originalValue = textureImporter.isReadable;
+
+            if (texture == null)
+            {
+                Debug.LogWarning($"Could not load texture at {textureImporter.assetPath}");
+                return false;
+            }
+
+            // Skip non-2D textures (which don't support GetPixels)
+            if (!(texture is Texture2D texture2D))
+                return false;
+
+            // Skip textures which are child assets (fonts, embedded textures, etc.)
+            if (!AssetDatabase.IsMainAsset(texture))
+                return false;
+
+            // For non-readable textures, make it readable to use some functions (GetPixels())
+            if (textureImporter.isReadable)
+            {
+                if (IsSolidColor(texture2D))
```

Comment:
It should be a little bit faster to first check the dimensions (the next if-statement) and then finally calling IsSolidColor().

So we would just skip the small 1x1 textures.